learning_resources/utils.py

```diff
@@ -473,74 +475,108 @@ def offeror_delete_actions(offeror: LearningResourceOfferor):
     hook.offeror_delete(offeror=offeror)
 
 
-def _walk_ocw_topic_map(
-    topics: dict, parent: None | LearningResourceTopic = None
-) -> None:
+def _walk_topic_map(topics: list, parent: None | LearningResourceTopic = None) -> None:
     """
     Walk the topic map provided and create topic records accordingly.
 
     This will recursively walk through the topics list and create/update topic
-    records as appropriate. There's just names here so if the record exists with
-    the same name, it'll update; otherwise, it creates.
+    records as appropriate. The topic records are stored in this format:
+    - name - the name of the topic
+    - id - the ID for the topic
+    - icon - the icon we should display for the topic (a Remixicon, generally)
+    - mappings - mappings for topics found in offeror data
+    - children - child topics (records in this same format)
 
     Args:
-    - topics (dict): the topics to process
+    - topics (list of dict): the topics to process
     - parent (None or LearningResourceTopic): the parent topic (for inner loops)
     Returns:
     - None
     """
 
     for topic in topics:
-        lr_topic, _ = LearningResourceTopic.objects.filter(name=topic).update_or_create(
+        lr_topic, created = LearningResourceTopic.objects.filter(
+            Q(name=topic["name"]) | Q(id=topic["id"])
+        ).update_or_create(
             defaults={
```

Comment:
What do you think of changing this to:
```python
        ).update_or_create(
            id=topic["id"],
            defaults={
```

Assuming the file will always have an id supplied, anyway.  This way the topics will be assigned the ids in the yaml file, and if running `upsert_topic_data_file` on a file that specifies an existing id and a changed name, the name will be updated on the existing topic instead of a new topic being created and the id value being ignored.

If `id` isn't always present, then maybe that arg could be used conditionally.

Reply:
The ID isn't always there: you may be adding a new one. However, this lead me down a bit of a rabbit hole wrt the IDs - this was using database (int) IDs and that ended up being a problem, so I refactored to add a UUID field. (Using database IDs ended up causing weirdness because inserting rows with IDs doesn't update the identity sequence, which generates database errors when you're trying to add new ones.)